openshift-kubecost-install.md

```diff
@@ -0,0 +1,513 @@
+```
+- Create K8s secret name dbsecret:
+
+```Bash
+kubectl create secret generic dbsecret \
+    --namespace kubecost \
+    --from-file=USERNAME \
+    --from-file=PASSWORD
+```
+- Verify if the credentials appears correctly - Optional (Any trailing space or new line etc ...)
+
+```Bash
+kubectl -n kubecost get secret dbsecret -o json | jq '.data | map_values(@base64d)'
+```
+**5. (optional): Configure Kubecost recording rules for Grafana Cloud using cortextool**
+
+To set up recording rules in Grafana Cloud, download the [cortextool CLI utility](https://github.com/grafana/cortex-tools). While they are optional, they offer improved performance.
+
+After installing the tool, create a file called \_kubecost\_rules.yaml\_ with the following command:
+
+<details>
+  <summary>Click to see code</summary>
+
+```yaml
+cat << EOF > kubecost-rules.yaml
+namespace: "kubecost"
+groups:
+  - name: CPU
+    rules:
+      - expr: sum(rate(container_cpu_usage_seconds_total{container_name!=""}[5m]))
+        record: cluster:cpu_usage:rate5m
+      - expr: rate(container_cpu_usage_seconds_total{container_name!=""}[5m])
+        record: cluster:cpu_usage_nosum:rate5m
+      - expr: avg(irate(container_cpu_usage_seconds_total{container_name!="POD", container_name!=""}[5m])) by (container_name,pod_name,namespace)
+        record: kubecost_container_cpu_usage_irate
+      - expr: sum(container_memory_working_set_bytes{container_name!="POD",container_name!=""}) by (container_name,pod_name,namespace)
+        record: kubecost_container_memory_working_set_bytes
+      - expr: sum(container_memory_working_set_bytes{container_name!="POD",container_name!=""})
+        record: kubecost_cluster_memory_working_set_bytes
+  - name: Savings
+    rules:
+      - expr: sum(avg(kube_pod_owner{owner_kind!="DaemonSet"}) by (pod) * sum(container_cpu_allocation) by (pod))
+        record: kubecost_savings_cpu_allocation
+        labels:
+          daemonset: "false"
+      - expr: sum(avg(kube_pod_owner{owner_kind="DaemonSet"}) by (pod) * sum(container_cpu_allocation) by (pod)) / sum(kube_node_info)
+        record: kubecost_savings_cpu_allocation
+        labels:
+          daemonset: "true"
+      - expr: sum(avg(kube_pod_owner{owner_kind!="DaemonSet"}) by (pod) * sum(container_memory_allocation_bytes) by (pod))
+        record: kubecost_savings_memory_allocation_bytes
+        labels:
+          daemonset: "false"
+      - expr: sum(avg(kube_pod_owner{owner_kind="DaemonSet"}) by (pod) * sum(container_memory_allocation_bytes) by (pod)) / sum(kube_node_info)
+        record: kubecost_savings_memory_allocation_bytes
+        labels:
+          daemonset: "true"
+EOF
+```
+</details>
+
+Then, making sure you are in the same directory as your \_kubecost\_rules.yaml\_, load the rules using `cortextool`. Replace the address with your Grafana Cloud’s Prometheus endpoint (Remember to omit the /api/prom path from the endpoint URL).
```

Comment:
Not sure where `\_` is from, but based on the code sample above, it looks like it should be kubecost-rules.yaml
```suggestion
Then, making sure you are in the same directory as your `kubecost-rules.yaml`, load the rules using `cortextool`. Replace the address with your Grafana Cloud’s Prometheus endpoint (Remember to omit the /api/prom path from the endpoint URL).
```